cpp/csp/adapters/websocket/WebsocketEndpoint.h

```diff
@@ -0,0 +1,79 @@
+#ifndef _IN_CSP_ADAPTERS_WEBSOCKETS_ENDPOINT_H
+#define _IN_CSP_ADAPTERS_WEBSOCKETS_ENDPOINT_H
+
+// need a base -> TLS, base -> No TLS
+#include <csp/adapters/websocket/ClientInputAdapter.h>
+#include <websocketpp/config/asio_client.hpp>
+#include <websocketpp/config/asio_no_tls_client.hpp>
+#include <websocketpp/client.hpp>
+#include <functional>
+
+namespace csp::adapters::websocket {
+
+using on_message_cb = std::function<void(std::string)>;
+using on_send_fail_cb = std::function<void(const std::string&)>;
+using void_cb = std::function<void()>;
+
+using message_ptr = websocketpp::config::core_client::message_type::ptr;
+
+
+class WebsocketEndpointBase {
+public:
+    WebsocketEndpointBase(csp::Dictionary properties);
+
```

Comment:
previous comment wasnt addressed, should have a virtual dtor

Reply:
for got to commit, will change